src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -500,29 +514,109 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
         const language =
             workspace.textDocuments.find((document) => document.uri.toString() === this.owner?.toString())
                 ?.languageId ?? PYTHON_LANGUAGE;
-        const notebookCell = new NotebookCellData(
+        const notebookCellData = new NotebookCellData(
             isMarkdown ? NotebookCellKind.Markup : NotebookCellKind.Code,
             strippedCode,
             isMarkdown ? MARKDOWN_LANGUAGE : language
         );
-        notebookCell.metadata = { interactiveWindowCellMarker };
-
+        notebookCellData.metadata = { interactiveWindowCellMarker };
         edit.replaceNotebookCells(
             notebookDocument.uri,
             new NotebookRange(notebookDocument.cellCount, notebookDocument.cellCount),
             [
-                notebookCell // TODO generalize to arbitrary languages and cell types
+                notebookCellData // TODO generalize to arbitrary languages and cell types
             ]
         );
         await workspace.applyEdit(edit);
 
+        const notebookCell = notebookDocument.cellAt(notebookDocument.cellCount - 1);
+        let result = true;
+
         // Request execution
-        await this.commandManager.executeCommand('notebook.cell.execute', {
-            ranges: [{ start: notebookDocument.cellCount - 1, end: notebookDocument.cellCount }],
-            document: notebookDocument.uri,
-            autoReveal: true
-        });
-        return true;
+        if (!debugInfo) {
+            await this.commandManager.executeCommand('notebook.cell.execute', {
+                ranges: [{ start: notebookDocument.cellCount - 1, end: notebookDocument.cellCount }],
+                document: notebookDocument.uri,
+                autoReveal: true
+            });
+        } else {
+            const notebook = this.kernel?.notebook;
```

Comment:
Just a minor suggestion so feel free to ignore. But this function was already longish, so adding in the debugger case (which is quite long) makes the function and standard non-debug case a bit harder to read and follow, might be nice to subfunction it.